lib/utils/string_extension.dart

```diff
@@ -47,4 +47,30 @@ extension StringCasingExtension on String {
   String toTomMatrixId() {
     return '@$this:tom-dev.xyz';
   }
+
+  List<String> getMentionsFromMessage() {
+    final RegExp regex = RegExp(r"@\[([^\]]+)\]");
+    final Iterable<Match> matches = regex.allMatches(this);
+
+    final List<String> mentions = [];
+    for (final Match match in matches) {
+      final String? mention = match.group(0);
+      if (mention != null) {
+        mentions.add(mention);
+      }
+    }
+    return mentions;
+  }
+
+  List<String> getAllMentionedUserIdsFromMessage(Room room) {
+    final List<String> mentionUserIds = [];
+    for (final String mention in getMentionsFromMessage()) {
+      final String? userId = room.lastEvent!.room.getMention(mention);
```

Comment:
When i firstly create the room, does the room have any event yet

Reply:
I'm not sure, I'll use nullable "?" instead